src/data/battler-tags/damaging-trap-tag.ts

```diff
@@ -44,7 +44,7 @@ export abstract class DamagingTrapTag extends TrappedTag {
   }
 
   override canAdd(pokemon: Pokemon): boolean {
-    return !pokemon.getTag(...TRAPPED_BATTLER_TAG_TYPES) && !pokemon.getTag(BattlerTagType.SUBSTITUTE);
+    return !pokemon.hasTag(...TRAPPED_BATTLER_TAG_TYPES) && !pokemon.hasTag(BattlerTagType.SUBSTITUTE);
```

Comment:
```suggestion
    return !pokemon.hasTag(...TRAPPED_BATTLER_TAG_TYPES, BattlerTagType.SUBSTITUTE);
```
I think these can be merge?